### ISSUE TITLE
Develop a logging facade 

### ISSUE THREAD

**Issue by hferentschik** [minishift/minishift#169]
Logging seems to be a bit all over the place in Minishift. There are ftm.Print statements as well we usages of glog. On top of this used libraries like for example libmachine have their own logging wrapper. 

Obviously the target for ftm vs glog is different, but if nothing else we should document a logging strategy. Personally I think it should even be possible to somehow have some unifying/configurable wrapper.

**Comment by hferentschik** [comment]
If nothing else we should document when and how to log in the dev guide. From a users perspective it would make sense on document how to enable the various logging capabilities.

**Comment by budhrg** [comment]
Some design philosophy on logging by William Kennedy https://www.goinggo.net/2017/05/design-philosophy-on-logging.html

**Comment by budhrg** [comment]
Found one package https://github.com/sirupsen/logrus . Seems to be maintained often.

**Comment by hferentschik** [comment]
This issue is not about introducing a new logging dependency (at least not if we don't have to). It is about unifying how we do logging, how the log format looks like, how we can configure  it and how we maybe can introduce logging categories.

**Comment by anjannath** [comment]
Based on @hferentschik [comment.](https://github.com/minishift/minishift/issues/169#issuecomment-311740188) we need to decide the following.

-  [ ] Logging format (Structure, what information in logs).
-  [ ] Logging levels / categories.
-  [ ] Logging pkg to use. (logrus, glog, go-logging, zap, etc..)

**Comment by anjannath** [comment]
We (@budhrg and /me) looked into the code and found out that, _minishift_ does not use that much of logging, the logging is mostly spit out by _libmachine_. Since there is not much logging in _minishift_, i think its best to understand what is expected out of this. Are we going to handle everything on the stdout using a logging framework? that means the msgs from preflight checks are also handled by the logging framework.

As for frameworks, _libmachine_ uses its own logging [pkg](https://github.com/docker/machine/tree/master/libmachine/log). In minishift we are using **glog**, which allows us to have different levels of logging using the `-v` flag. If we use the log pkg from _libmachine_ then our logs will come together with _libmachine_ logs and we won't have levels anymore. We also need to decide what kind of information belongs to which level, AFAIK right now there is no criteria as to which msg belong to which level of verbosity.

/cc @minishift/minishift-dev

**Comment by gbraad** [comment]
Well, as in the original description, many logs are provided as `fmt.Print`, which can be wrapped as a logging level, either info, error, etc. this has to be reviewed and decided upon.

One of those changes can be to move pre-flight checks to info, and ONLY error out on an actual error.

**Comment by anjannath** [comment]
> Well, as in the original description, many logs are provided as fmt.Print, which can be wrapped as a logging level, either info, error, etc. this has to be reviewed and decided upon.

Yes, we do have a lot of `fmt.Print`, +1 for having a discussion around this.

**Comment by anjannath** [comment]
After brief discussion on irc about this with @gbraad couple of things came up. Right now we don't log the startup settings, we should be logging that, also the logging to files inside `~/.minishift/logs/` are not done properly.

We show TOO much information regarding the progress of `minishift start` as preflight checks, we could clean that up too, and only show information to user only when something fails.

**Comment by gbraad** [comment]
* allow logging to a file (either happens to general or profile)
  * allow to log to stdout and stderr
  * allow to set different log levels
    * error
    * info
    * warning
  * possibly allow to increase log level during execution (^1)


^1 - Imagine just showing `-- Starting Minishift: 'cluster02' ................ OK`, but during execution, you could imagine that some steps might take long or never finish. Pressing `ESC` might increase the loglevel, so the current step is shown.

**Comment by budhrg** [comment]
> Pressing ESC might increase the loglevel, so the current step is shown.

This is interesting. So, instead of passing via start command option, we increase the log level by pressing `ESC` ? 
How, we decide what to show as part of it?

**Comment by anjannath** [comment]
Yes, so, we won't show any logs just, `Starting Minishift ............................... OK`
Then if the process is stuck and the user wants to see the progress of the current task, he can press `ESC` to get the logs.

> How, we decide what to show as part of it?

That we have to decide, my guess would be all the logs in the info level.
